File: app/src/main/java/me/instabattle/app/settings/State.java
package me.instabattle.app.settings;

import me.instabattle.app.models.Battle;
import me.instabattle.app.models.User;

// FIXME
public class State {
    public static Battle chosenBattle;

    public static User currentUser = new User("wackloner");
    public static String token = null;

    public static boolean creatingBattle;
}

File: app/src/main/kotlin/me/instabattle/app/settings/State.kt
package me.instabattle.app.settings

import com.chibatching.kotpref.KotprefModel
import me.instabattle.app.models.Battle
import me.instabattle.app.models.User

// FIXME
object State {
    var chosenBattle: Battle? = null

    lateinit var currentUser: User
    var creatingBattle: Boolean = false
}

object KState: KotprefModel() {
    var token by stringPref()
    var creatingBattle by booleanPref()
}

/*object UserInfo : KotprefModel() {
    var gameLevel by enumValuePref(GameLevel.NORMAL)
    var name by stringPref()
    var code by nullableStringPref()
    var age by intPref(default = 14)
    var highScore by longPref()
    var rate by floatPref()
    val prizes by stringSetPref {
        val set = TreeSet<String>()
        set.add("Beginner")
        return@stringSetPref set
    }
}

enum class GameLevel {
    EASY,
    NORMAL,
    HARD
}*/
